patches/tModLoader/Terraria/ModLoader/Core/TmodFile.cs

```diff
@@ -436,4 +448,40 @@ private void Upgrade()
 		Open();
 		// Read contract fulfilled
 	}
+
+	internal bool VerifyHash()
```

Comment:
Should probably be something like `VerifyHash() => hashVerified ??= _VerifyHash()` where `_VerifyHash` has the impl and is `private`.